File: Homework/Quang_Minh/HomeWorkLesson7Part2/FindBottomLeftTreeValue.java
package HomeWorkLesson7Part2;

import HomeWorkLesson7.TreeNode;

import java.util.ArrayDeque;
import java.util.Queue;

public class FindBottomLeftTreeValue {
    public int findBottomLeftValue(TreeNode root) {
        Queue<TreeNode> queue = new ArrayDeque<TreeNode>();
        queue.add(root);
        int leftValue = 0;
        while (true) {
            if (queue.size() == 0) break;
            int size = queue.size();
            for (int i = 0; i < size; i++) {
                TreeNode currentNode = queue.poll();
                if (i == 0 && currentNode.left == null && currentNode.right == null) leftValue = currentNode.val;
                if (currentNode.left != null) queue.add(currentNode.left);
                if (currentNode.right != null) queue.add(currentNode.right);
            }

        }
        return leftValue;
    }
}

File: Homework/Van_Le/lesson7_continue/FindBottomLeftTreeValue/FindBottomLeftTreeValue.cs
namespace LeetCodeSolutions
{
    public class FindBottomLeftTreeValue
    {
        public int FindBottomLeftValue(TreeNode root)
        {
            if(root == null) return -1;

            Queue<TreeNode> queue = new Queue<TreeNode>();
            queue.Enqueue(root);
            int ans = -1;

            while(queue.Count > 0)
            {
                int size = queue.Count;
                for(int i = 0; i < size; i++)
                {
                    TreeNode node = queue.Dequeue();

                    if (i == size - 1) ans = node.val;

                    if (node.right != null) queue.Enqueue(node.right);
                    if (node.left != null) queue.Enqueue(node.left);
                }
            }

            return ans;
        }
    }
}
